### PR TITLE
[Merged by Bors] - feat(RingTheory/Localization): more API and kernels of localized maps

### PR DESCRIPTION
**Pull request by chrisflav**
This PR adds various API lemmas on localization. In particular it computes the kernel of a localized map as the localization of the kernel.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by erdOne** [comment]
Thanks!
maintainer merge

File: Mathlib/RingTheory/Ideal/Maps.lean [hunk 1]
```diff
@@ -908,3 +908,14 @@ lemma coe_ideal_map (I : Ideal A) :
     Ideal.map f I = Ideal.map (f : A →+* B) I := rfl
 
 end AlgHom
+
+namespace Algebra
+
+variable {R : Type*} [CommSemiring R] (S : Type*) [Semiring S] [Algebra R S]
+
+@[simps!]
+def idealMap (I : Ideal R) : I →ₗ[R] I.map (algebraMap R S) :=
```
Comment (riccardobrasca) [inline Mathlib/RingTheory/Ideal/Maps.lean:917]: The linter will ask for a documentation here.

**Review by riccardobrasca — APPROVED** [review]
Thanks!

bors d+

File: Mathlib/RingTheory/Ideal/Maps.lean (hunk 1 above)
Comment (chrisflav) [reply Mathlib/RingTheory/Ideal/Maps.lean:917]: That got lost in copy pasting. Thanks!

**Comment by chrisflav** [comment]
Thanks for the reviews!

**Pull request closed without merging**